client/components/UserDecalPanel.js

```diff
@@ -87,25 +109,31 @@ class UserDetailsPanel extends Component {
     return null
   }
 
+  renderDecalCode = (decal) => (
+    <div className="decal" key={decal.id}>
+      <div className="code">
+        <button
+          aria-label="Show decal code"
+          className="icon toggle"
+          name={decal.id}
+          onClick={this._handleToggle}
+          type="button">
+          <FontAwesomeIcon icon={this.state.decalsVisible[decal.id] ? 'eye' : 'eye-slash'} fixedWidth />
+        </button>
+        {this.state.decalsVisible[decal.id] ? decal.attributes.code : `•••••-•••••-•••••-•••••-${decal.attributes.code.substring(24)}`}
+      </div>
+      <div className="claimed-at">{formatAsEliteDate(decal.attributes.claimedAt)}</div>
+    </div>
+  )
+
+  renderDecalCodes = () => {
+    const { decals } = this.props
 
-
-
-
-  /***************************************************************************\
-    Public Methods
-  \***************************************************************************/
-
-  async componentDidMount () {
-    const {
-      checkDecalEligibility,
-      eligible,
-    } = this.props
-
-    if (!eligible) {
-      await checkDecalEligibility()
+    if (!decals.length) {
+      return null
     }
 
-    this.setState({ checkingEligibility: false })
+    return decals.map(this.renderDecalCode)
```

Comment:
you could actually write this bit out as 
```suggestion
    return Boolean(decals.length) && decals.map(this.renderDecalCode)
```
 and remove that `if (!decals.length)` block above.